### ISSUE TITLE
How to configure and read log_query from multiple active clickhouse clients

### ISSUE THREAD

**Issue by healiseu** [yandex/ClickHouse#3790]
Hi, I have several connected clients to my clickhouse-server, e.g. clickhouse-client, dbeaver database manager, python mymarilyn/clickhouse-driver.

1. **Is there an SQL command to display active connections to clickhouse-server ? How do I view connected clients ?**

2. **I am sending queries from all these clients and I want to read information from the `query_log` table**

For example, read the 10 most recent queries:

```sql
SELECT
    user,
    client_hostname AS host,
    client_name,
    type,
    formatDateTime(query_start_time, '%T') as started,
    query_duration_ms AS msec,
    round(memory_usage/1048576) AS MEM_MB,
    result_rows AS RES_CNT,
    toDecimal32(result_bytes/1048576, 6) AS RES_ΜΒ,
    read_rows AS R_CNT,
    round(read_bytes/1048576) AS R_MB,
    written_rows AS W_CNT,
    round(written_bytes/1048576) AS W_MB
FROM system.query_log
WHERE type=2
ORDER BY started DESC
LIMIT 10;

DESCRIBE TABLE system.query_log
```
I searched documentation on [`query_log`][1] and `log_queries` but it is still at a premature stage. I am trying to understand what is this `type` field  and how the following settings affect the logged queries ? 
```sql
SELECT name, value
FROM system.settings 
WHERE name LIKE '%log%'

┌─name──────────────────────┬─value──┐
│ log_queries               │ 0      │
│ log_queries_cut_to_length │ 100000 │
│ log_profile_events        │ 1      │
│ log_query_settings        │ 1      │
│ log_query_threads         │ 1      │
│ send_logs_level           │ none   │
└───────────────────────────┴────────┘
```
Are these true/false values ? After `set log_queries = 1` 
I ran several queries from my clients but only those from `clickhouse-client` were recorded in the query_log table. 

#### How can I get the last N queries sent from all my clients with a query similar to the one I wrote above ?

[1]: https://clickhouse.yandex/docs/en/operations/server_settings/settings/#query_log

**Comment by filimonov** [comment]
1. Use `show processlist`, or `select * from system.processes`
2. `log_queries` is a user/session level setting. You can enable it in a session by sending `set log_queries = 1`; or for the particular user by editing `users.xml` section; or for default user profile if you want to enable that globally.

**Comment by filimonov** [comment]
>  I am trying to understand what is this type field 

Please check: https://github.com/yandex/ClickHouse/issues/847#issuecomment-306013716 


> I am trying to understand ... how the following settings affect the logged queries ?

```sql
SELECT *
FROM system.settings 
WHERE name LIKE '%log%'

┌─name──────────────────────┬─value──┬─changed─┬─description────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ log_queries               │ 0      │       0 │ Log requests and write the log to the system table.                                                                                                                │
│ log_queries_cut_to_length │ 100000 │       0 │ If query length is greater than specified threshold (in bytes), then cut query when writing to query log. Also limit length of printed query in ordinary text log. │
│ log_profile_events        │ 1      │       0 │ Log query performance statistics into the query_log and query_thread_log.                                                                                          │
│ log_query_settings        │ 1      │       0 │ Log query settings into the query_log.                                                                                                                             │
│ log_query_threads         │ 1      │       0 │ Log query threads into system.query_thread_log table. This setting have effect only when 'log_queries' is true.                                                    │
│ send_logs_level           │ none   │       0 │ Send server text logs with specified minumum level to client. Valid values: 'trace', 'debug', 'information', 'warning', 'error', 'none'                            │
└───────────────────────────┴────────┴─────────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

6 rows in set. Elapsed: 0.004 sec. 
```

**Comment by healiseu** [comment]
Hi @filimonov  thanks for the reply, I was a bit confused with the session/user/global settings. I have edited users.xml for my user and I confirm that I was able to get query log for the last N queries. It also seems the type setting I used is the most appropriate in my case. According to @alexey-milovidov

1.  - successful start of query execution
2.  - successful end of query execution
3.  - exception before start of query execution
4. - exception while query execution

Each query create one or two events in query_log table:

* two events with types 1 and 2: usual, successful case;
* two events with types 1 and 4;
* single event with type 3;

It will help others if log_query is properly documented with an example like the one above ;-)
